sys/kern/src/arch/arm_m.rs

```diff
@@ -1269,25 +1286,30 @@ pub fn irq_status(n: u32) -> abi::IrqStatus {
 
     // See if the interrupt is enabled by checking the bit in the Interrupt Set
     // Enable Register.
-    let enabled = nvic.iser[reg_num].read() & bit_mask == bit_mask;
+    let iser_reg = nvic.iser.get(reg_num).ok_or(UsageError::NoIrq)?;
+    let enabled = iser_reg.read() & bit_mask == bit_mask;
     status.set(abi::IrqStatus::ENABLED, enabled);
 
     // See if the interrupt is pending by checking the bit in the Interrupt
     // Set Pending Register (ISPR).
     let pending = nvic.ispr[reg_num].read() & bit_mask == bit_mask;
     status.set(abi::IrqStatus::PENDING, pending);
 
-    status
+    Ok(status)
 }
 
-pub fn pend_software_irq(InterruptNum(n): InterruptNum) {
+pub fn pend_software_irq(
+    InterruptNum(n): InterruptNum,
+) -> Result<(), UsageError> {
```

Comment:
I was about to say "hmm, shouldn't we update the documentation for the `software_irq` KIPC to reflect that it will fault you if you ask for a nonexistent IRQ...but then, I realized that the Hubris reference docs for KIPCs [doesn't even mention `software_irq`, and instead suggests that `find_faulted_task` is KIPC 8](https://hubris.oxide.computer/reference/#_find_faulted_task_8), which it isn't (it's KIPC 9, and `software_irq` is 8). This is almost certainly my fault back when I added the software IRQ KIPC, so I should probably fix that...